ivy.el

```diff
@@ -981,7 +981,19 @@ If the text hasn't changed as a result, forward to `ivy-alt-done'."
                 (if (and ivy--directory
                          (not (eq (ivy-state-history ivy-last)
                                   'grep-files-history)))
-                    (expand-file-name ivy-text ivy--directory)
+                    (if (and (string= ivy-text "")
+                             (eq (ivy-state-collection ivy-last)
+                                 #'read-file-name-internal))
+                        ;; For `read-file-name' compat, unchanged initial input
+                        ;; means that `ivy-read' shall return INITIAL-INPUT.
+                        ;; `read-file-name-default' `string-equal' return value
+                        ;; with provided INITIAL-INPUT to detect that the user
+                        ;; choose the default, `default-filename'.
```

Comment:
Nitpick: if `default-filename` is referring to the eponymous `read-file-name` argument, then I suggest formatting it as `DEFAULT-FILENAME`, as is already done with `INITIAL-INPUT`.

Reply:
Not quite. IICR, `default-filename` refers to the `(setq default-filename ... )` value in `read-file-name-default`. Still, good point and the comment wording should be edited and rephrased.

Reply:
> Not quite. IICR, `default-filename` refers to the `(setq default-filename ... )` value in `read-file-name-default`.

Okay, but that local variable still corresponds to its enclosing function's positional argument of the same name, no?

> Still, good point and the comment wording should be edited and rephrased.

That you added a comment at all is already very useful, but I am always grateful for efforts in improving documentation, thanks!

Reply:
> Okay, but that local variable still corresponds to its enclosing function's positional argument of the same name, no?

Yup, here the focus was more on the value to be returned, not necessarily the provided value, that's all. I have no problem with an upcase.

Reply:
> Yup, here the focus was more on the value to be returned, not necessarily the provided value, that's all.

Gotcha. In that case I suggest explicitly referring to the function's return value, rather than to some implementation detail.